mq/plugins/stackdriver.py

```diff
@@ -0,0 +1,68 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+# Copyright (c) 2017 Mozilla Corporation
+
+import urllib
+from platform import node
+from mozdef_util.utilities.toUTC import toUTC
+
+
+class message(object):
+    def __init__(self):
+        """
+            Plugin used to fix object type discretions with cloudtrail messages
+        """
+        self.registration = ["pubsub"]
+        self.priority = 5
+
+        try:
+            self.mozdefhostname = "{0}".format(node())
+        except:
+            self.mozdefhostname = "failed to fetch mozdefhostname"
+            pass
+
+    def onMessage(self, message, metadata):
+        # trust no one mr mulder
+        if "tags" not in message:
+            return (message, metadata)
+        if "pubsub" not in message["tags"]:
+            return (message, metadata)
+        if "details" not in message:
+            return (message, metadata)
+
+        event = message["details"]
+
+        if "logName" not in event:
+            return (message, metadata)
+        else:
+            # XXX: implement filtering of audit types that we want to see (yaml)
+            newmessage = dict()
+            logtype = urllib.parse.unquote(event["logName"]).split("/")[-1].strip()
```

Comment:
Is there any chance that `event["logName"]` isn't this format? Should we verify that before we try and access the last element in the array? `[-1]`?

Reply:
I added a check for the field's presence, but I do not expect the format to change here. It is part of official event definition and Google seems to keep those intact. In other words, if this changes than it most likely means we have to write our plugins again, because they changed  data structures for everything they have.